packages/private/src/walkObject.ts

```diff
@@ -11,7 +11,7 @@ export function walkObject<T extends Walkable, MapTo>(
   fn: (value: Primitive, path: Array<string>) => MapTo,
   path: Array<string> = [],
 ): MapLeafNodes<T, MapTo> {
-  const clone = obj.constructor();
+  const clone = {} as any;
```

Comment:
The types are a bit gnarly here, `clone` was `any` before this change, so this just keeps the status quo.